packages/http/src/stamp.ts

```diff
@@ -1,38 +1,137 @@
-import * as crypto from "crypto";
+import { webcrypto } from "crypto";
 
 // Specific byte-sequence for curve prime256v1 (DER encoding)
 const PRIVATE_KEY_PREFIX = Buffer.from(
   "308141020100301306072a8648ce3d020106082a8648ce3d030107042730250201010420",
   "hex"
 );
 
-export function stamp(input: {
+export async function stamp(input: {
   content: string;
   publicKey: string;
   privateKey: string;
 }) {
   const { content, publicKey, privateKey } = input;
 
-  const privateKeyBuffer = Buffer.from(privateKey, "hex");
+  const key = await importPrivateKey(privateKey);
+  const signature = await signMessage(key, content);
+
+  return {
+    publicKey: publicKey,
+    scheme: "SIGNATURE_SCHEME_TK_API_P256",
+    signature: signature,
+  };
+}
+
+async function importPrivateKey(
+  privateKeyHex: string
+): Promise<webcrypto.CryptoKey> {
+  const privateKeyBuffer = Buffer.from(privateKeyHex, "hex");
   const privateKeyPkcs8Der = Buffer.concat([
     PRIVATE_KEY_PREFIX,
     privateKeyBuffer,
   ]);
 
-  const privateKeyObject = crypto.createPrivateKey({
-    type: "pkcs8",
-    format: "der",
-    key: privateKeyPkcs8Der,
-  });
+  return await webcrypto.subtle.importKey(
+    "pkcs8",
+    privateKeyPkcs8Der,
+    {
+      name: "ECDSA",
+      namedCurve: "P-256",
+    },
+    false, // not extractable
+    ["sign"] // allow signing
+  );
+}
 
-  const sign = crypto.createSign("SHA256");
-  sign.write(Buffer.from(content));
-  sign.end();
-  const signature = sign.sign(privateKeyObject, "hex");
+async function signMessage(
+  privateKey: webcrypto.CryptoKey,
+  content: string
+): Promise<string> {
+  const signatureIeee1363 = await webcrypto.subtle.sign(
+    {
+      name: "ECDSA",
+      hash: "SHA-256",
+    },
+    privateKey,
+    Buffer.from(content)
+  );
 
-  return {
-    publicKey: publicKey,
-    scheme: "SIGNATURE_SCHEME_TK_API_P256",
-    signature: signature,
-  };
+  const signatureDer = convertEcdsaIeee1363ToDer(
+    new Uint8Array(signatureIeee1363)
+  );
+
+  return Buffer.from(signatureDer).toString("hex");
+}
+
+/**
+ * `SubtleCrypto.sign(...)` outputs signature in IEEE P1363 format:
+ * - https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign#ecdsa
+ *
+ * Turnkey expects the signature encoding to be DER-encoded ASN.1:
+ * - https://github.com/tkhq/tkcli/blob/7f0159af5a73387ff050647180d1db4d3a3aa033/src/internal/apikey/apikey.go#L149
+ *
+ * Code modified from https://github.com/google/tink/blob/6f74b99a2bfe6677e3670799116a57268fd067fa/javascript/subtle/elliptic_curves.ts#L114
+ *
+ * Transform an ECDSA signature in IEEE 1363 encoding to DER encoding.
+ *
+ * @param ieee the ECDSA signature in IEEE encoding
+ * @return ECDSA signature in DER encoding
+ */
+function convertEcdsaIeee1363ToDer(ieee: Uint8Array): Uint8Array {
```

Comment:
I think some basic tests are in order for this function? I know it's indirectly tested by the high-level "signature is valid" test, but I'd feel better with a few unit tests given this function has pretty complex, hard-to-review-at-a-glance logic.